### PR TITLE
board: arm: ubx_bmd340eval_nrf52840

### PR DESCRIPTION
**Pull request by ghost**
Add support for u-blox BMD-340-EVAL EVK for the BMD-340-EVAL
based on nRF52840

Tested with samples/basic/blinky, samples/basic/button,
and samples/bluetooth/peripheral_hr

Signed-off-by: Bob Recny <bob.recny@u-blox.com>

### CODE REVIEW

File: boards/arm/ubx_bmd340eval_nrf52840/ubx_bmd340eval_nrf52840_defconfig [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+# Copyright (c) 2021 u-blox AG
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_SOC_SERIES_NRF52X=y
+CONFIG_SOC_NRF52840_QIAA=y
+CONFIG_BOARD_UBX_BMD340EVAL_NRF52840=y
+
+# Enable MPU
+CONFIG_ARM_MPU=y
```
Comment (ioannisg) [inline boards/arm/ubx_bmd340eval_nrf52840/ubx_bmd340eval_nrf52840_defconfig:11]: Enable HW_STACK_PROTECTION=y as well, here.

**Review by ioannisg — APPROVED** [review]

**Comment by brec-u-blox** [comment]
I am running into failures with the tests. See https://buildkite.com/zephyr/zephyr/builds/20763.
The board I'm attempting to add is nearly identical to the `nRF52840DK_nRF52840` board, except we're using our own module which contains the nRF52840 IC. I see in the tests directory that there are three specific mentions of the Nordic board:
```
~/zephyrproject/zephyr/tests$ grep CONFIG_BOARD_NRF52840DK_NRF52840 -r *
drivers/uart/uart_async_api/src/test_uart.h:#if defined(CONFIG_BOARD_NRF52840DK_NRF52840)
drivers/uart/uart_mix_fifo_poll/src/main.c:#if defined(CONFIG_BOARD_NRF52840DK_NRF52840)
drivers/adc/adc_api/src/test_adc.c:	defined(CONFIG_BOARD_NRF52840DK_NRF52840) || \
```
Do I add `CONFIG_BOARD_BMD340EVAL_NRF52840` to these files?
If not, what is the best practice?

**Comment by ioannisg** [comment]
@brec-u-blox could you fix CI issues (or rebase to re-trigger Ci build)?

**Comment by brec-u-blox** [comment]
Something got fouled up in the PR - 273 commits is not correct. Closing. I'll learn some more and try again.

**Pull request closed without merging**